projects/fleet-management-grid/src/fleet-management-grid.ts

```diff
@@ -1,81 +1,1048 @@
+
+  }
+
+  /* Event Handlers */
+  private locationEventHandler = (event: any) => {
+    const path = event.composedPath();
+
+    const clickedInsideOverlay = path.includes(this.locationOverlay);
+    const clickedOnTrigger = this.lastOverlayTrigger ? path.includes(this.lastOverlayTrigger) : false;
+
+    if (!clickedInsideOverlay && !clickedOnTrigger) {
+      this.closeLocationOverlay();
+    }
+  };
+
+  private driverEventHandler = (event: any) => {
+    const path = event.composedPath();
+
+    const clickedInsideOverlay = path.includes(this.driverOverlay);
+    const clickedOnTrigger = this.lastOverlayTrigger ? path.includes(this.lastOverlayTrigger) : false;
+
+    if (!clickedInsideOverlay && !clickedOnTrigger) {
+      this.closeDriverOverlay();
+    }
+  };
+
+  /* Close */
+  private closeLocationOverlay() {
+    if (this.locationOverlay) {
+      const overlay = this.locationOverlay;
+
+      overlay.classList.remove('visible');
+      this.overlayBackdrop.classList.remove("visible");
+
+      overlay.addEventListener('transitionend', () => {
+        overlay.style.display = 'none';
+      }, { once: true })
+    }
+  
+    document.removeEventListener("mousedown", this.locationEventHandler);
+    window.removeEventListener("wheel", this.locationEventHandler, true);
+  }
+
+  private closeDriverOverlay() {
+    if (this.driverOverlay) {
+      const overlay = this.driverOverlay;
+
+      overlay.classList.remove('visible');
+      this.overlayBackdrop.classList.remove("visible");
+
+      overlay.addEventListener('transitionend', () => {
+        overlay.style.display = 'none';
+      }, { once: true })
+    }
+  
+    document.removeEventListener("mousedown", this.driverEventHandler);
+    window.removeEventListener("wheel", this.driverEventHandler, true);
+  }
+
+
+  /* Map Logic */
+  private addSeriesWith(locations: any[], brush: string) {
+    const symbolSeries = new IgcGeographicSymbolSeriesComponent();
+    symbolSeries.dataSource = locations;
+    symbolSeries.latitudeMemberPath = "latitude";
+    symbolSeries.longitudeMemberPath = "longitude";
+    symbolSeries.markerBrush  = "White";
+    symbolSeries.markerOutline = brush;
+    symbolSeries.markerTemplate = {
+      measure: (measureInfo: DataTemplateMeasureInfo) => {
+        measureInfo.width = 24;
+        measureInfo.height = 24;
+      },
+      render: (renderInfo: DataTemplateRenderInfo) => {
+        const ctx = renderInfo.context;
+        const x = renderInfo.xPosition;
+        const y = renderInfo.yPosition;
+
+        const img = new Image();
+        img.src = 'location_pin.svg';
+        img.onload = () => {
+          ctx.drawImage(img, x - 12, y - 12, 32, 32);
+        };
+      }
+    } as IgDataTemplate;
+    this.map.series.add(symbolSeries);
+  }
+
+  /** Data Helpers */
+
+  private getStatusType(status: string): string {
+    const types: Record<string, string> = {
```

Comment:
This types are recreated on each template render and rerender. You could extract them into the models folder and just use them as imported type